Phoenix/PHSpace.m

```diff
@@ -222,9 +237,35 @@ - (void)removeWindows:(NSArray<PHWindow *> *)windows {
                                (__bridge CFArrayRef) @[@(self.identifier)]);
 }
 
+/**
+ * - https://github.com/kasper/phoenix/issues/348
+ * - https://github.com/koekeishiya/yabai/issues/2240#issuecomment-2116326165
+ * - https://github.com/ianyh/Amethyst/issues/1643#issuecomment-2132519682
+ */
+- (void)moveWindowsWithCompatId:(NSArray<PHWindow *> *)windows {
+    NSArray<NSNumber *> *ids = [self identifiersForWindows:windows];
+    NSUInteger windowCount = [ids count];
+    NSMutableData *windowIdSequence = [[NSMutableData alloc] initWithCapacity:windowCount * sizeof(CGWindowID)];
+    for (id item in ids) {
+        CGWindowID value = [item unsignedIntValue];
+        [windowIdSequence appendBytes:&value length:sizeof(CGWindowID)];
+    }
+
+    CGSConnectionID connection = CGSMainConnectionID();
+    CGSSpaceSetCompatID(connection, self.identifier, MoveWindowsCompatId);
+    CGSSetWindowListWorkspace(connection, (CGWindowID *)[windowIdSequence bytes], windowCount, MoveWindowsCompatId);
+    CGSSpaceSetCompatID(connection, self.identifier, 0x0);
+}
+
 - (void)moveWindows:(NSArray<PHWindow *> *)windows {
-    CGSMoveWindowsToManagedSpace(
-        CGSMainConnectionID(), (__bridge CFArrayRef)[self identifiersForWindows:windows], self.identifier);
+    if (![NSProcessInfo isOperatingSystemAtLeastSonoma145]) {
+        CGSMoveWindowsToManagedSpace(
+            CGSMainConnectionID(), (__bridge CFArrayRef)[self identifiersForWindows:windows], self.identifier);
+        return;
+    }
+
+    // CGSMoveWindowsToManagedSpace is broken in MacOS 14.5, so use the legacy Compat ID API instead.
```

Comment:
Then move this comment before the conditional.

```suggestion
    // CGSMoveWindowsToManagedSpace is broken in macOS 14.5, so use the legacy Compat ID API instead
```

Reply:
As mentioned in https://github.com/kasper/phoenix/pull/350#discussion_r1632003998, I think keeping the order better aligns with your intent, but adopted the comment style changes 👀